configuration.go

```diff
@@ -0,0 +1,156 @@
+package raft
+
+import "fmt"
+
+// ServerSuffrage determines whether a Server in a Configuration gets a vote.
+type ServerSuffrage int
+
+const (
+	// Voter is a server whose vote is counted in elections and whose match index
+	// is used in advancing the leader's commit index.
+	Voter ServerSuffrage = 1
```

Comment:
The problem with that is, since these are written into the log, we should be careful to never renumber or reuse numbers. It's the same reason why protocol buffers makes you assign numbers manually. So there's my rationale. I'll change to iota if you still think that's better.